Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -31,7 +31,12 @@ extension MongoCollection {
             return nil
         }
 
-        return InsertOneResult(insertedId: try document.getValue(for: "_id"))
+        guard let insertedId = try document.getValue(for: "_id") else {
+            // This case should not ever really happen, since we handle it above and give the document an _id.
+            throw MongoError.invalidArgument(message: "Failed to get value for _id from document")
```

Comment:
I don't think is an appropriate use of `MongoError.invalidArgument`, since it's a logical assertion occurring _after_ we've executed the operation and doesn't have much to do with the method's input arguments.

In PHP, I would throw an [UnexpectedValueException](https://github.com/mongodb/mongo-php-library/blob/1.4.2/src/Operation/FindAndModify.php#L221) for such a case, which is a [type of runtime exception](http://php.net/unexpectedvalueexception) (it depends on a result only known at runtime). The best equivalent in the Swift driver at the moment is probably `MongoError.commandError`, since you're using that when `distinct()` has a malformed response. Alternatively, a `preconditionFailure()` may be preferable here since a missing `_id` at this point means that either the driver has done something very wrong (i.e. the logic earlier in this same method hasn't done its job).

Reply:
I'll switch to a `preconditionFailure`, since this should _never_ happen.